### PR TITLE
refactor: Move SQLConnector into a separate module, for use by both SQLStream and SQLSink

### LINKED ISSUE #1090: refactor: Move `SQLConnector` into a separate module, for use by both `SQLStream` and `SQLSink`
Currently the `SQLConnector` class is in `stream` module in the same file as `SQLStream`, however it is also needed by `SQLSink` from the `sink` module. Moving the connector into a `sql` module would make it easier to work on sinks, and also provide a place for other shared, SQL-related code.

### PR DESCRIPTION
**Pull request by kgpayne**
Closes #1090 

- #1090 

----
:books: Documentation preview :books:: https://meltano-sdk--1091.org.readthedocs.build/en/1091/

### CODE REVIEW

**Comment by kgpayne** [comment]
This commit doesn't change any logic. I simply moved the `SQLConnector` into a new `sql` module 🙂

File: singer_sdk/streams/__init__.py [hunk 1]
```diff
@@ -3,12 +3,6 @@
...
+from singer_sdk.streams.sql import SQLStream
 
-__all__ = [
-    "Stream",
-    "GraphQLStream",
-    "RESTStream",
-    "SQLStream",
-    "SQLConnector",
-]
+__all__ = ["Stream", "GraphQLStream", "RESTStream", "SQLStream"]
```
Comment (aaronsteers) [inline singer_sdk/streams/__init__.py:8]: @kgpayne

I think we could go either way on this but are we sure we want to remove the reference of `singer_sdk.streams.SQLConnector`?

I'm totally okay with the decision to do so, but probably we should also remove `singer_sdk.sinks.SQLConnector` here if so: https://github.com/meltano/sdk/blob/kgpayne/issue1090/singer_sdk/sinks/__init__.py

I'd also be okay leaving it as referenceable from both `singer_sdk.streams` and `singer_sdk.sinks`. Small benefit of not breaking existing code with the slight downside of possible confusion of where to best import from.

cc @edgarrmondragon

File: singer_sdk/sql/__init__.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+"""Module for SQL-related operations."""
+
+from .connector import SQLConnector
```
Comment (aaronsteers) [inline singer_sdk/sql/__init__.py:3]: @kgpayne - It feels just a little odd to me that `SQLStream` and `SQLSink` are not in the `sql` module. 

What do you think of reversing the module ref name, from `singer_sdk/sql/connector.py` to `singer_sdk/connectors/sql.py` - which then would make `SQLConnector` importable from `singer_sdk.connectors` and also `singer_sdk.connectors.sql`. Not a ton of other uses for this namespace as of now, but this leaves open the possibility us us creating something like `singer_sdk.connectors.OpenAPIConnector` and similar.

Alternatively, we could put this under `singer_sdk/helpers/sql.py`.

What do you think?

cc @edgarrmondragon, @cjohnhanson

File: singer_sdk/streams/__init__.py (hunk 1 above)
Comment (edgarrmondragon) [reply singer_sdk/streams/__init__.py:8]: @kgpayne @aaronsteers

TLDR I'm okay with removing `SQLConnector` from the "public API" of the module.

---

`__all__` only affects _star_ imports.

For example, consider the module

```python
# mymod.py
__all__ = ["X", "Y"]

X = 1
Y = 2
Z = 3
```

Then the module's "public API" is only reflected when importing with `*`:

```python
>>> from mymod import *
>>> X, Y, Z
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'Z' is not defined
```

But the `Z` member can still be imported explicitly:

```python
>>> from mymod import X, Y, Z
>>> X, Y, Z
(1, 2, 3)
```

File: singer_sdk/sql/__init__.py (hunk 2 above)
Comment (edgarrmondragon) [reply singer_sdk/sql/__init__.py:3]: I really like `singer_sdk/connectors/sql.py`. It's similar to what I imagined for https://github.com/meltano/sdk/issues/317 😄

File: singer_sdk/streams/__init__.py (hunk 1 above)
Comment (aaronsteers) [reply singer_sdk/streams/__init__.py:8]: > `__all__` only affects _star_ imports.

Agreed, and understood. I believe it may also affect IDE auto-complete for those editors which will use it to hint at members within those modules, yes?

This SO article seems to confirm: 

https://stackoverflow.com/a/35710527/4298208

> Documentation tools
>
> Documentation and code autocompletion tools may (in fact, should) also inspect the `__all__` to determine what names to show as available from a module.

I do think it's worth being careful about out `__all__` declarations, for the purpose of those IDE hints.

Also - to clarify, I believe the 'breaking' part would be the prior references at the `__init__.py` level, specifically `singer_sdk.[streams|sinks].SQLConnector` - but this would still work `singer_sdk.[streams|sinks].sql.SQLConnector` because both `.../sql.py` files import `SQLConnector` at the top of the file.

Comment (kgpayne) [reply singer_sdk/streams/__init__.py:8]: Done ✅

**Review by edgarrmondragon — APPROVED** [review]
